can/interfaces/socketcand/socketcand.py

```diff
@@ -13,12 +13,96 @@
 import socket
 import time
 import traceback
+import urllib.parse as urlparselib
+import xml.etree.ElementTree as ET
 from collections import deque
+from typing import List
 
 import can
 
 log = logging.getLogger(__name__)
 
+DEFAULT_SOCKETCAND_DISCOVERY_ADDRESS = ""
+DEFAULT_SOCKETCAND_DISCOVERY_PORT = 42000
+
+
+def detect_beacon():
+    sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
```

Comment:
Use a context manager to automatically close the socket.

Reply:
Done in 8e4a6db